com.avaloq.tools.ddk.check.core/src/com/avaloq/tools/ddk/check/resource/CheckBatchLinkableResourceStorageFacade.java

```diff
@@ -1,14 +1,52 @@
 package com.avaloq.tools.ddk.check.resource;
 
+import java.io.IOException;
+import java.util.Collections;
+import java.util.NavigableSet;
+
+import org.eclipse.emf.common.util.URI;
+import org.eclipse.emf.ecore.resource.ResourceSet;
 import org.eclipse.xtext.generator.IFileSystemAccess;
 import org.eclipse.xtext.generator.IFileSystemAccessExtension3;
+import org.eclipse.xtext.resource.persistence.ResourceStorageLoadable;
 import org.eclipse.xtext.resource.persistence.StorageAwareResource;
+import org.eclipse.xtext.util.RuntimeIOException;
 import org.eclipse.xtext.xbase.resource.BatchLinkableResourceStorageFacade;
 
+import com.avaloq.tools.ddk.check.runtime.configuration.IModelLocation;
+import com.avaloq.tools.ddk.check.runtime.registry.ICheckCatalogRegistry;
+
 
 @SuppressWarnings("restriction")
 public class CheckBatchLinkableResourceStorageFacade extends BatchLinkableResourceStorageFacade {
 
+  /**
+   * Returns a generated URI corresponding to the location on disk of
+   * the binary model associated with the provided {@param resource}.
+   *
+   * @param resource
+   *          the .check file resource whose binary model we are looking for
+   * @return the URI of the binary model storage for the provided {@param resource}
+   */
+  protected URI getBinaryStorageURI(final StorageAwareResource resource) {
+    String resourceName = computeOutputPath(resource);
+
+    ICheckCatalogRegistry checkCatalogRegistry = ICheckCatalogRegistry.INSTANCE;
+    NavigableSet<IModelLocation> checkModelLocations = checkCatalogRegistry.getAllCheckModelLocations();
+
+    String modelLocationQuery = getSourceContainerURI(resource).toString();

```

Comment:
Maybe just
 String modelLocationQuery = getSourceContainerURI(resource).toString().replace("platform:/resource/", ""); //$NON-NLS-1$ //$NON-NLS-2$

to avoid reassigning the local variable.